pkg/apis/core/validation/validation.go

```diff
@@ -3010,6 +3010,32 @@ func validateInitContainers(containers []core.Container, otherContainers []core.
 	return allErrs
 }
 
+func validateHostUsers(spec *core.PodSpec, volumes map[string]core.VolumeSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.UserNamespacesSupport) {
+		if spec.SecurityContext != nil && spec.SecurityContext.HostUsers != nil {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("hostUsers"), "field not supported. Enable the featureGate UserNamespacesSupport"))
+		}
+		return allErrs
+	}
+
+	// Only check if hostUsers is false (userns is enabled).
```

Comment:
New suggestion
```suggestion
	// Only check volume type if hostUsers is false (userns is enabled).
```

(We should make it clear what thing is being checked)

Reply:
@sftim I don't see the point. If in the future we need to check more things, that comment will easily become either out of date or a small burden to maintain.

What we want to say, IMHO, is really what this comment says: We only make the following checks if this field is set to false, otherwise there isn't anything else to check.